### PR TITLE
Add startlxqt

### PR DESCRIPTION
**Pull request by Besanon**

### CODE REVIEW

**Comment by roddhjav** [comment]
You didn't moved it, you copied it, so now, there is the same profile at two location.

File: apparmor.d/groups/lxqt/startlxqt [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+include <tunables/global>
+
+@{exec_pathstlx} = @{bin}/startlxqt
+profile startlxqt @{exec_pathstlx} {
+  include <abstractions/base>
+  include <abstractions/consoles>
+  include <abstractions/freedesktop.org>
+  include <abstractions/fontconfig-cache-read>
+  include <abstractions/lxqt>
+  include <abstractions/X-strict>
```
Comment (roddhjav) [inline apparmor.d/groups/lxqt/startlxqt:17]: `freedesktop.org` and `X-strict` are included in `lxqt`, remove them

File: apparmor.d/groups/lxqt/startlxqt [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+# Copyright (C) 2023 Alexandre Pujol <alexandre@pujol.io>
+# Copyright (C) 2024 Besanon  <m231009ts@mailfence.com>
+# SPDX-License-Identifier: GPL-2.0-only
+
+abi <abi/4.0>,
+
+include <tunables/global>
+
+@{exec_pathstlx} = @{bin}/startlxqt
+profile startlxqt @{exec_pathstlx} {
```
Comment (roddhjav) [inline apparmor.d/groups/lxqt/startlxqt:11]: This always need to be named: `@{exec_path}`
```suggestion
@{exec_path} = @{bin}/startlxqt
profile startlxqt @{exec_path} {
```

File: apparmor.d/groups/lxqt/startlxqt [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+  @{bin}/systemctl  rCx -> systemctl,
+  @{bin}/dbus-update-activation-environment  rCx -> dbus,
+
+  /usr/share/color-schemes/{,**} r,
+  /usr/share/desktop-directories/{,**} r,
+  /usr/share/kservices5/{,**} r,
+  /usr/share/mime/{,**} r,
+
+  /etc/locale.alias r,
```
Comment (roddhjav) [inline apparmor.d/groups/lxqt/startlxqt:38]: Remove this rule. It is part of the base abstraction.

File: apparmor.d/groups/lxqt/startlxqt [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+  /etc/machine-id r,
+  /etc/xdg/menus/{,**} r,
+
+  @{HOME}/ r,
+  owner @{HOME}/.Xauthority r,
```
Comment (roddhjav) [inline apparmor.d/groups/lxqt/startlxqt:43]: Remove, it is already included in `X-strict`

File: apparmor.d/groups/lxqt/startlxqt [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+
+  owner @{user_cache_dirs}/ rw,
```
Comment (roddhjav) [inline apparmor.d/groups/lxqt/startlxqt:45]: Remove, included in `lxqt` (through `xdg-desktop`)

**Comment by roddhjav** [comment]
What display manager are you using?

You need to add `include if exists <local/startlxqt>` as last rule of the profile (cf the failed CI). I would recommend you to locally run `make check` before submitting PR so that you can verify yourself against these checks.

File: apparmor.d/groups/lxqt/startlxqt [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+  owner @{user_share_dirs}/kservices5/{,**} r,
+  owner @{user_share_dirs}/sddm/wayland-session.log rw,
+  owner @{user_share_dirs}/sddm/xorg-session.log rw,
+
+  owner /tmp/#@{int} rw,
+  owner /tmp/startlxqt.@{rand6} rwl -> /tmp/#@{int},
+
+  owner @{run}/user/@{uid}/ r,
+
+        @{PROC}/sys/kernel/core_pattern r,
```
Comment (roddhjav) [inline apparmor.d/groups/lxqt/startlxqt:62]: Remove this rule as it is part of the base abs.

**Comment by Besanon** [comment]
Hello Alex,

maybe you already noticed: after upgrading to apparmor 4.0 in Arch linux there is a conflict shown with

abstractions/authentication :

The line witch unix_chkpwd  is culprit for a duplicate x stetement e.g with sudo

Am 22.10.2024 um 21:25, Alex ***@***.***> schrieb:

@roddhjav commented on this pull request.
In apparmor.d/groups/lxqt/startlxqt:> +        @{user_cache_dirs}/ksycoca5_* rwkl -> @{user_cache_dirs}/#@{int},++  owner @{user_config_dirs}/#@{int} rw,+  owner @{user_config_dirs}/lxqt/ rw,+  owner @{user_config_dirs}/menus/{,**} r,++  owner @{user_share_dirs}/kservices5/{,**} r,+  owner @{user_share_dirs}/sddm/wayland-session.log rw,+  owner @{user_share_dirs}/sddm/xorg-session.log rw,++  owner /tmp/#@{int} rw,+  owner /tmp/startlxqt.@{rand6} rwl -> /tmp/#@{int},++  owner @{run}/user/@{uid}/ r,++        @{PROC}/sys/kernel/core_pattern r,
Remove this rule as it is part of the base abs.
—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you authored the thread.Message ID: ***@***.***>

-- 
Sent with https://mailfence.com  
Secure and private email

**Comment by roddhjav** [comment]
Just update apparmor.d, this was expected, it should be fixed with #560.

**Comment by Besanon** [comment]
oh, havent seen that before, just delete the second instance

File: apparmor.d/groups/lxqt/startlxqt [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+  owner /tmp/startlxqt.@{rand6} rwl -> /tmp/#@{int},
+
+  owner @{run}/user/@{uid}/ r,
+
+  owner @{PROC}/@{pid}/maps r,
+
+  /dev/tty rw,
+  /dev/tty@{int} rw,
+
+ include if exists <local/startlxqt>
```
Comment (roddhjav) [inline apparmor.d/groups/lxqt/startlxqt:62]: You need to indent it. (Please check the result of the CI).

**Pull request merged**